data_diff/diff_tables.py

```diff
@@ -124,13 +105,31 @@ def _relevant_columns(self) -> List[str]:
             + list(self.extra_columns)
         )
 
+    def count(self) -> Tuple[int, int]:
+        return self.database.query(self._make_select(columns=[Count()]), int)
```

Comment:
The reason I didn't include this was because I was worried that someone might accidentally call this in `diff_tables`, which slows the whole thing down... This might be a work injury from working on big codebases where this _will_ happen, so I'm OK if you call me over-cautious and keep it. You can also make the argument that if we have a proper benchmark suite that's the better fix.

It just feels a bit weird to me that this method is _only_ there for testing

Reply:
I think of it as API for the users too. Once they create a TableSegment, they can count to see how much it catches.

I'm not too worried that we'll call it by accident. Also it's a method now and not a property, which signifies (int terms of semantics) that it might be an expensive operation.

But I don't have a strong feeling about this, so I'm open to change my mind.

Reply:
Let's just keep it